src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs

```diff
@@ -265,5 +265,50 @@ protected unsafe void SetLengthCore(long value)
                 _filePosition = value;
             }
         }
+
+        public override int Read(byte[] buffer, int offset, int count) => ReadSpan(new Span<byte>(buffer, offset, count));
+
+        public override int Read(Span<byte> buffer) => ReadSpan(buffer);
+
+        private unsafe int ReadSpan(Span<byte> destination)
+        {
+            if (_fileHandle.IsClosed)
+            {
+                ThrowHelper.ThrowObjectDisposedException_FileClosed();
+            }
+            else if ((_access & FileAccess.Read) == 0)
+            {
+                ThrowHelper.ThrowNotSupportedException_UnreadableStream();
+            }
+
+            int r = RandomAccess.ReadAtOffset(_fileHandle, destination, _filePosition, _path);
```

Comment:
Per my previous comment, changing FileStream in this manner should make it easy for us to do a perf comparison.  Have you already run the numbers (sync Read/Write perf on a FileStream created as async)?